File: selenium_java/pages/CarritoPagina.java
// POM - página del carrito
package pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import java.time.Duration;
import java.util.List;
import java.util.stream.Collectors;

public class CarritoPagina {
    private WebDriver driver;
    private WebDriverWait espera;

    public CarritoPagina(WebDriver driver) {
        this.driver = driver;
        this.espera = new WebDriverWait(driver, Duration.ofSeconds(5));
    }

    public int obtenerCantidadProductos() {
        espera.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector(".cart_item")));
        return driver.findElements(By.cssSelector(".cart_item")).size();
    }

    public List<String> obtenerNombresProductos() {
        espera.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector(".cart_item")));
        List<WebElement> items = driver.findElements(By.cssSelector(".cart_item .inventory_item_name"));
        return items.stream().map(WebElement::getText).collect(Collectors.toList());
    }

    public String obtenerNombrePrimerProducto() {
        return driver.findElement(By.cssSelector(".inventory_item_name")).getText();
    }
}

File: selenium_python/paginas/CarritoPagina.py
# POM - página del carrito de SauceDemo
from selenium.webdriver.common.by import By
from selenium.webdriver.support.ui import WebDriverWait
from selenium.webdriver.support import expected_conditions as EC


class CarritoPagina:
    def __init__(self, driver):
        self.driver = driver
        self.espera = WebDriverWait(driver, 5)

    def obtener_cantidad_productos(self):
        self.espera.until(EC.presence_of_element_located((By.CSS_SELECTOR, ".cart_item")))
        return len(self.driver.find_elements(By.CSS_SELECTOR, ".cart_item"))

    def obtener_nombres_productos(self):
        self.espera.until(EC.presence_of_element_located((By.CSS_SELECTOR, ".cart_item")))
        items = self.driver.find_elements(By.CSS_SELECTOR, ".cart_item .inventory_item_name")
        return [item.text for item in items]

    def obtener_nombre_primer_producto(self):
        return self.driver.find_element(By.CSS_SELECTOR, ".inventory_item_name").text
